build/scripts/obtain_dotnet.sh

```diff
@@ -27,7 +15,16 @@ install_dotnet () {
     local DOTNET_VERSION="$(get_tool_version dotnetSdk)"
     local DOTNET_PATH="${THIS_DIR}"/../../Binaries/dotnet-cli
 
-    if [[ ! -x "${DOTNET_PATH}/dotnet" ]]
+    # check if the correct `dotnet` is already on the PATH
+    if command -v dotnet >/dev/null 2>&1
+    then
+        if [[ "${FORCE_DOWNLOAD:-false}" != true && "$(dotnet --version)" = "${DOTNET_VERSION}" ]]
```

Comment:
Note: `dotnet --version` will fail with an exit code and bad message if the version differs from that required in global.json. 

https://github.com/dotnet/sdk/issues/1673

Not sure there is much you can do here though. Think we may need to keep this for now and wait for the bug fix in the linked bug to catch up with us.